wrapanapi/msazure.py

```diff
@@ -359,8 +359,50 @@ def list_stack(self, resource_group=None, days_old=0):
     def list_flavor(self):
         raise NotImplementedError('list_flavor not implemented.')
 
-    def list_network(self):
-        raise NotImplementedError('list_network not implemented.')
+    def list_networks(self):
+        self.logger.info("Attempting to List Azure Virtual Private Networks of the provider region")
+        # Azure API returns all networks from all regions, and there is options to filter by region.
+        # In CFME only the networks of the provider regions are displayed.
+        all_networks = self.network_client.virtual_networks.list_all()
+        networks_in_region = []
+        location = self.region.replace(' ', '').lower()
+        for network in all_networks:
+            if network.location == location:
+                networks_in_region.append(network.name)
+        return networks_in_region
+
+    def list_subnets(self):
+        self.logger.info("Attempting to List Azure Subnets")
+        # There is no way to list all the subnets from a network filtered by location, and there
+        # is only one network in the resource_group defined in cfme_data.
+        all_networks = self.network_client.virtual_networks.list_all()
+        location = self.region.replace(' ', '').lower()
+        subnets = []
+        for network in all_networks:
+            if network.location == location:
+                for subnet in network.subnets:
+                    subnets.append(subnet.name)
```

Comment:
Looks like this needs to be de-duped?

```python
if subnet.name not in subnets:
    subnets.append(subnet.name)
```
Current behavior is:
```
In [16]: m.list_subnets()
Out[16]: 
[u'default',
 u'default',
 u'default',
 u'default',
 u'test-tfischer-subnet',
 u'Subnet-1']
```

Reply:
Are you speaking about the 'default' subnets_networks?
If yes it is normal that there is many with this name, CFME displays also 4 defaults networks. This is why there is this loop.

Reply:
Ok, now I see in the portal that you can have have same subnet name w/in different networks. I'm not sure what we are using this list_subnets method for, but how do we know which 'default' is which? :) Should we return the name of the virtual network along with the subnet name too?